haskell/test/Test/Ligo/TreasuryDAO.hs

```diff
@@ -76,31 +76,20 @@ validProposal = uncapsNettest $ withFrozenCallStack do
   withSender (AddressResolved owner1) $
     call dao (Call @"Propose") (ProposeParams proposalSize proposalMeta)
 
-  checkTokenBalance (frozenTokenId) dao owner1 166
-  checkTokenBalance (unfrozenTokenId) dao owner1 34
+  checkTokenBalance frozenTokenId dao owner1 366
 
```

Comment:
Connected to the previous point: AFAIU this results now changed because there's only transfers of `frozenTokenId` going on, however I think we'll need to have a function similar to `checkTokenBalance` to fetch the status of a `dummyFA2` instead here.

All things considered, as far as tests are concerned, we might just want to originate a dummy FA2 contract every time we originate a DAO, just so to have it available in the tests themselves. What do you think?

Reply:
>fetch the status of a dummyFA2 instead here

Here the proposal is not actually flushed, so the transfer does not actually happen. The only transfer call the token contract get is the token transfer for `freeze` call. And the test AFAIU tests that we can raise a valid proposal. So is there any point in checking the token contract state here. We have other tests for BaseDAO contract that check that the transfer in proposal is actually executed by checking the token contract state to ensure that it received a transfer call as expected.

>we might just want to originate a dummy FA2 contract every time we originate a DAO, just so to have it available in the tests themselves. What do you think?

We are actually doing that right now and the `OriginateFn` type alias has been changed to include the FA2 Contract's address in the tuple it returns. 

https://github.com/tqtezos/baseDAO/blob/sras/%23170-byo-extension/haskell/test/Test/Ligo/BaseDAO/Common.hs#L44

Reply:
Ah I was confused and didn't check properly, thanks for the explanation.

> And the test AFAIU tests that we can raise a valid proposal. So is there any point in checking the token contract state here

I agree, no point in doing so, let's keep them like this :ok_hand: